assets/js/atomic/blocks/product-elements/price/edit.js

```diff
@@ -1,133 +1,51 @@
 
-const TextControl = ( {
-	fontSize,
-	setFontSize,
-	color,
-	setColor,
-	colorLabel,
-} ) => (
-	<>
-		<FontSizePicker value={ fontSize.size } onChange={ setFontSize } />
-		{ /* ColorPalette doesn't accept an id. */
-		/* eslint-disable-next-line @wordpress/no-base-control-with-label-without-id */ }
-		<BaseControl label={ colorLabel }>
-			<ColorPalette
-				value={ color.color }
-				onChange={ setColor }
-				label={ __( 'Color', 'woo-gutenberg-products-block' ) }
-			/>
-		</BaseControl>
-	</>
-);
-const PriceEdit = ( {
-	fontSize,
-	saleFontSize,
-	setFontSize,
-	setSaleFontSize,
-	color,
-	saleColor,
-	setColor,
-	setSaleColor,
-	attributes,
-	setAttributes,
-} ) => {
-	const { align } = attributes;
+const PriceEdit = ( { attributes, setAttributes } ) => {
+	const blockProps = useBlockProps();
 	return (
 		<>
-			{ isFeaturePluginBuild() && (
-				<BlockControls>
+			<BlockControls>
+				{ isFeaturePluginBuild() && (
 					<AlignmentToolbar
-						value={ align }
-						onChange={ ( nextAlign ) => {
-							setAttributes( { align: nextAlign } );
+						value={ attributes.textAlign }
+						onChange={ ( newAlign ) => {
+							setAttributes( { textAlign: newAlign } );
 						} }
 					/>
-				</BlockControls>
-			) }
-			<InspectorControls>
-				{ isFeaturePluginBuild() && (
-					<>
-						<PanelBody
-							title={ __(
-								'Price',
-								'woo-gutenberg-products-block'
-							) }
-						>
-							<TextControl
-								color={ color }
-								setColor={ setColor }
-								fontSize={ fontSize }
-								setFontSize={ setFontSize }
-								colorLabel={ __(
-									'Color',
-									'woo-gutenberg-products-block'
-								) }
-							/>
-						</PanelBody>
-						<PanelBody
-							title={ __(
-								'Sale price',
-								'woo-gutenberg-products-block'
-							) }
-						>
-							<TextControl
-								color={ saleColor }
-								setColor={ setSaleColor }
-								fontSize={ saleFontSize }
-								setFontSize={ setSaleFontSize }
-								colorLabel={ __(
-									'Color',
-									'woo-gutenberg-products-block'
-								) }
-							/>
-						</PanelBody>
-					</>
 				) }
-			</InspectorControls>
-			<Block { ...attributes } />
+			</BlockControls>
+			<div { ...blockProps }>
+				<Block { ...attributes } />
+			</div>
 		</>
 	);
 };
 
-const Price = isFeaturePluginBuild()
-	? compose( [
-			withFontSizes( 'fontSize' ),
-			withFontSizes( 'saleFontSize' ),
-			withFontSizes( 'originalFontSize' ),
-			withColors( 'color', { textColor: 'color' } ),
-			withColors( 'saleColor', { textColor: 'saleColor' } ),
-			withColors( 'originalColor', { textColor: 'originalColor' } ),
-			withProductSelector( {
-				icon: BLOCK_ICON,
-				label: BLOCK_TITLE,
-				description: __(
-					'Choose a product to display its price.',
-					'woo-gutenberg-products-block'
-				),
-			} ),
-	  ] )( PriceEdit )
-	: PriceEdit;
-
-export default Price;
+export default withProductSelector( {
```

Comment:
I think that we should use the HOC `withProductSelector` only when `isFeaturePluginBuild()` is true

Reply:
I think this should be always available as this block does nothing without product present.

Reply:
Mmm, but in this case, we are introducing a new block in Core. I guess that this is not the scope of this PR.

If I search `Product Price` is available as a block with our plugin, but this is not true with WooCommerce Core only.

![image](https://user-images.githubusercontent.com/4463174/155752573-d2d69c78-f96f-4226-ac65-6207d35be4d7.png)
*with the plugin*

![image](https://user-images.githubusercontent.com/4463174/155752627-0b8c0dab-f487-4268-8497-3c0a0c23f832.png)
*without plugin*

Reply:
Ah, I thought it was responsible only for adding product selector to the block toolbar hmm. How is it that it controls if block is selectable or not? Not sure If I follow how this HOC  in Edit controls works 😢 

(also your first screenshot is showing a bug with icons for blocks eek! see filter on 1st vs 2nd)

Reply:
After our investigation, we found that this check is not done here, but [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/d03169059e73b6fa8813ab48003bb6a11f40b1c9/assets/js/atomic/blocks/product-elements/shared/config.tsx#L32).

So it is a safe change even because other blocks don't have this check (for example `Product Image` and  `Product Rating`)